frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -335,10 +352,59 @@ function DataFrame({
     }
   }, [element.formId, resetEditingState, widgetMgr])
 
+  const freezeColumns = isEmptyTable
+    ? 0
+    : columns.filter((col: BaseColumn) => col.isIndex).length
+
+  // Determine if the table requires horizontal or vertical scrolling:
+  React.useEffect(() => {
+    if (resizableContainerRef.current && !isEmptyTable) {
+      // TODO(lukasmasuch): This is only a hacky and temporary solution until
+      // glide-data-grid provides a better way to determine this:
+      // https://github.com/glideapps/glide-data-grid/issues/784
+
+      // Get the bounds of the data grid scroll area:
+      const scrollAreaBounds = resizableContainerRef.current
+        ?.querySelector(".dvn-stack")
```

Comment:
What is the `dvn-stack`?

Reply:
the scroll area of data grid, I will add that to the comment.